frontend/src/lib/schemas/profileFormSchema.ts

```diff
@@ -1,19 +1,25 @@
-import { z } from "zod";
+import { z } from 'zod'
 
 export const profileFormSchema = z.object({
-	firstName: z.string({ message: "First name is required" }).nonempty(),
-	lastName: z.string({ message: "Last name is required" }).nonempty(),
-	title: z.string({message: "Please select a title"}).nonempty(),
-	preferredName: z.string().optional(),
-	dateOfBirth: z.string({message: "Please enter a valid date of birth"}).nonempty(),
-	gender: z.string({ message: "Please select from the options above"}).nonempty(),
-	email: z.string({ message: "Please enter a valid email" }).email(),
-	phone: z.string({message:"Please enter a valid phone number"}).nonempty(),
-	address: z.string({message: "Please enter a valid address"}).nonempty(),
-	suburb: z.string({message: "Please enter a valid suburb"}).nonempty(),
-	city: z.string({message: "Please enter a valid city"}).nonempty(),
-	postCode: z.coerce.number({ message: "Postcode must be at least 4 digits"}).min(1000)
-  .max(9999, "Postcode must be at most 4 digits")
-  .int("Postcode must be a number"),
-	country: z.string().nonempty(),
-});
\ No newline at end of file
+  firstName: z.string({ message: 'First name is required' }).nonempty(),
+  lastName: z.string({ message: 'Last name is required' }).nonempty(),
+  title: z.string({ message: 'Please select a title' }).nonempty(),
+  preferredName: z.string().optional(),
+  dateOfBirth: z
+    .string({ message: 'Please enter a valid date of birth' })
+    .nonempty(),
+  gender: z
+    .string({ message: 'Please select from the options above' })
+    .nonempty(),
+  email: z.string({ message: 'Please enter a valid email' }).email(),
+  phone: z.string({ message: 'Please enter a valid phone number' }).nonempty(),
+  address: z.string({ message: 'Please enter a valid address' }).nonempty(),
+  suburb: z.string({ message: 'Please enter a valid suburb' }).nonempty(),
+  city: z.string({ message: 'Please enter a valid city' }).nonempty(),
+  postCode: z
+    .string()
+    .min(4, { message: 'Postcode must be at least 4 characters' })
+    .regex(/^\d+$/, { message: 'Postcode must contain only digits' })
+    .max(4, 'Postcode must be at most 4 digits'),
```

Comment:
This comment is sort of an aside, you don't actually need to change this at all, I just thought it's neat.

Since you're looking for a postcode to be exactly four digits long you could just check `length` rather than both `min` and `max`

```suggestion
    .string()
    .regex(/^\d+$/, { message: 'Postcode must contain only digits' })
    .length(4, 'Postcode must be 4 digits'),
```

But since you can check it against regex you can actually be sneaky and include the length check within the regex itself. Rather than using `+` to repeat any number of times you can use `{4}` to specify you're checking for exactly 4 digits.

Usually I wouldn't actually suggest doing something like this since it technically makes it a bit more difficult to read the regex (Most devs aren't great at writing regex and even worse at reading it) But I think here it's short enough that it wouldn't cause an actual issue.

```suggestion
    .string()
    .regex(/^\d{4}$/, { message: 'Invalid postcode' }),
```